### DISCUSSION TITLE
Taxes, import duties, shipping cost on PO

### DISCUSSION THREAD

**Discussion by jacky4566** [inventree/InvenTree#4234] [Ideas]
Is there plans to incorporate Taxes and Import Duties to Purchase Orders?

Tax on purchasing:
Right now I have to manually adjust each line to add 5% GST. It would be great to have this applied at a per line item with a settable default per supplier. 

Landed costs:
Secondly it would be awesome if Import Duties and Fright costs were added to the Line items, either by weight or price. 
Image of how this works in CIN7:
![image](https://user-images.githubusercontent.com/1008762/213806664-094f948e-39ac-43fd-8a21-4969f83f3b91.png)

**Comment by matmair** [comment]
@jacky4566 not currently and that would certainly take a while to implement right.

For shipping, taxes etc. we have additional line items currently (the table below the main items). You can use either the UI or the API to add the info. Depending on your level of comfort with python you could also add a [plugin](https://docs.inventree.org/en/latest/extend/plugins/) to automate that.

InvenTree does not cover any meaningful aspects of accounting so you will need to use some sort of ERP / account software for the foreseeable future so maybe that covers it?

**Comment by jacky4566** [comment]
Ok, so it sounds like I'm not missing something hidden. Yes you can add "additional line items"  but that doesn't seem to do anything except for reporting. 

IMO this is pretty core feature for calculating the real cost of production. Purchasing costs need to include taxes, duties, and fright otherwise what's the point of doing any pricing in InvenTree. While its not ERP / account software we should be able to calculate the real cost of produced goods after a build.

Please consider this an official feature request. 

I hope to dive deeper into the software and contribute to this project.

**Reply by matmair** [reply]
> Ok, so it sounds like I'm not missing something hidden. Yes you can add "additional line items" but that doesn't seem to do anything except for reporting.
> 
It changes the total cost of the order.

> 
> Please consider this an official feature request.
> 

We have feature request templates with mandatory fields, please [fill one out](https://github.com/inventree/InvenTree/issues/new?assignees=&labels=enhancement%2Ctriage%3Anot-checked&template=feature_request.yaml&title=%5BFR%5D+title). It helps to ensure that the developed feature matches the requested feature in the end.

I think we have different ways of pricing goods.
Do you include tax in your raw price calculations? At least in the EU you either do not need to pay it when purchasing for your business or can just subtract the value of the tax owed at the end of the financial cycle.
We calculate the average freight, handling etc. cost per part over the whole portfolio and apply that accordingly to each part. If doing it on a per-order basis the price for SKUs would always be changing - which is normally not appreciated by regular customers or wholesalers.

**Reply by SchrodingersGat** [reply]
@jacky4566 to add to the reply from @matmair - implementing this in a generic way which covers all use-cases would be very tricky and potentially require more financial knowledge depth than we currently have. Certainly we are willing to consider suggestions and/or PRs as submitted :)

**Comment by j03l** [comment]
Has there been any sort of development on adding taxes and duties to pricing?

**Reply by SchrodingersGat** [reply]
@j03l none!

**Reply by j03l** [reply]
Ah yep. How difficult would it be to create a plugin to apply an extra % (i.e %10) of value to the cost of each line on a goods received order, to only items in your default currency? What sort of Mixins would you recommend I use? 

Expected outcome: 

My default currency is AUD. Have a check box at the top of a goods received saying whether tax is in the cost or not. 

When a part is added (that is in AUD) to a goods received and the box is checked, add an additional 10% to the item when the form is sent. Else, remove 10% from each line. 

This means I wouldn't have to do any additional work in any other part of the user interface, nor add any additional database entries for including or excluding tax. 

This also means that all cost values in Inventree would be a real world figure. 😊
